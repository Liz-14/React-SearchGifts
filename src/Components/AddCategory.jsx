import PropTypes from 'prop-types'; // define tipo a las props
import { useState } from 'react'

export const AddCategory = ({onNewCategory}) => {

   const [inputValue, setInputValue] = useState('')

   const onInputChange = ({target}) => {
      setInputValue(target.value)
   }

   const onSubmit = (event) => {
      //evita el refresh al hacer submit en el form
      event.preventDefault()
      //Sale de la funci√≥n si se envian espacios vacios o 1 solo caracter
      if(inputValue.trim().length <= 1) return

      //setCategories(categories => [inputValue, ...categories])
      onNewCategory( inputValue.trim() )
      //Limpiar input
      setInputValue('')
   }

   return (
      <form onSubmit={onSubmit} aria-label = 'form'>
         <input 
            type="text" 
            placeholder="Buscar Gifs"
            value={inputValue}
            onChange={ onInputChange }
         />
      </form>
   )
}

AddCategory.propTypes = {
   onNewCategory: PropTypes.func.isRequired
}

