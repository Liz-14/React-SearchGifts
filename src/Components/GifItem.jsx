import PropTypes from 'prop-types'; // define tipo a las props

export const GifItem = ({ title, url }) => {
  return (
    <div className="card">
      <p>{ title }</p>
      <img src={ url } alt={ title }/>
    </div>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
