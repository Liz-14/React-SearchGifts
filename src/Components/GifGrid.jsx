import PropTypes from 'prop-types'; // define tipo a las props
import { GifItem } from './GifItem'
import { useFetchGifs } from '../Hooks/useFetchGifs'

export const GifGrid = ({category}) => {

  //custom hook
  const { images, isLoading } = useFetchGifs ( category )

  return (
    <>
      <h3>{ category }</h3>

      {
        /* isLoading ? <h2>Cargando...</h2> : null */
        isLoading && <h2>Cargando...</h2>
      }

      <div className="card-grid">
        {
          images.map(images => 
              <GifItem 
                key = { images.id }
                { ...images }
              />
            )
        }
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}