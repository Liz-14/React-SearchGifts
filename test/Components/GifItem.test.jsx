import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/Components/GifItem'

describe('Pruebas en <GifItem/>', () => {

   const title = 'Inuyasha'
   const url = 'https://media2.giphy.com/media/fLv2F5rMY2YWk/giphy.gif?cid=fdfb23a7de0tm48clioc2khvqvepu5lg7gflpl42ubrml3wo&rid=giphy.gif&ct=g'

   test('Debe de hacer match con el snapshot', () => {
      
      const {container} = render(<GifItem title={ title } url={ url } />);
      expect(container).toMatchSnapshot();

   });

   test('Debe mostrar la imagen con la URL y el ALT indicados:', () => {
      
      render(<GifItem title={ title } url={ url } />);
      //screen.debug()
      //expect(screen.getByRole('img').src).toBe(url);
      //expect(screen.getByRole('img').alt).toBe(title);

      const { src, alt } = screen.getByRole('img')
      expect(src).toBe(url)
      expect(alt).toBe(title)

   });

   test('Debe mostar el titulo en el componente', () => {
      
      render(<GifItem title={ title } url={ url } />);
      //Buscar el texto en el componente
      expect(screen.getByText(title)).toBeTruthy();

   });
});