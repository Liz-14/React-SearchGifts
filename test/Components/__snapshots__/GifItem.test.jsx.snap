// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <p>
      Inuyasha
    </p>
    <img
      alt="Inuyasha"
      src="https://media2.giphy.com/media/fLv2F5rMY2YWk/giphy.gif?cid=fdfb23a7de0tm48clioc2khvqvepu5lg7gflpl42ubrml3wo&rid=giphy.gif&ct=g"
    />
  </div>
</div>
`;
